pkg/querier/distributor_queryable.go

```diff
@@ -37,13 +41,25 @@ type distributorQuerier struct {
 	distributor Distributor
 	ctx         context.Context
 	mint, maxt  int64
+	streaming   bool
+	chunkIterFn chunkIteratorFunc
 }
 
 func (q *distributorQuerier) Select(sp *storage.SelectParams, matchers ...*labels.Matcher) (storage.SeriesSet, storage.Warnings, error) {
-	mint, maxt := q.mint, q.maxt
-	if sp != nil {
-		mint = sp.Start
-		maxt = sp.End
+	// Kludge: Prometheus passes nil SelectParams if it is doing a 'series' operation,
```

Comment:
Can we clarify who is kludging what?  I think it's important that Cortex is relying on a behaviour of Prometheus, to make a decision that helps minimise effort at runtime.

Reply:
I copied this comment from `querier` type, which had direct reference to distributor previously.

Any suggestion on how to improve/change this comment? It seems clear to me.

Reply:
Maybe we can just remove the `Kludge:`? :) All in all, we're just describing how Prometheus works and why we handle the special case of `sp == nil`.

Reply:
I don't see this documented anywhere else, so to me `Kludge:` indicates that this is possibly a hack that can break :)